### PR TITLE
add event summary creator to trim long summaries

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSBPS-486

### Change description ###

add event summary creator to trim long summaries

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventSummaryCreator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.util.List;
+
+public class EventSummaryCreator {
+
+    private static final String EVENT_SUMMARY_REPLACE = "{XX}";
+    private static final String EVENT_SUMMARY_TRIM_DOTS = "...";
+    private static String EVENT_SUMMARY_TEMP =
+        "Attaching exception record(%d) document numbers:" + EVENT_SUMMARY_REPLACE + " to case:%d";
+    private static int EVENT_SUMMARY_MAX_SIZE = 1024;
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventSummaryCreator.java:11]: Do we know this value?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventSummaryCreator.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        String eventSummary = String.format(
+            EVENT_SUMMARY_TEMP,
+            exceptionRecordId,
+            caseId
+        );
+
+        int remainingSize = EVENT_SUMMARY_MAX_SIZE - eventSummary.length();
+        String documentNumbers = documentNumberList.toString();
+        if (documentNumbers.length() > remainingSize + EVENT_SUMMARY_REPLACE.length()) {
+            documentNumbers = documentNumbers.substring(0, remainingSize) + EVENT_SUMMARY_TRIM_DOTS;
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventSummaryCreator.java:31]: Shouldn't it be 
`documentNumbers.substring(0, remainingSize - EVENT_SUMMARY_TRIM_DOTS.length())`
?

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventSummaryCreator.java:31]: in the length there is EVENT_SUMMARY_REPLACE length's as well so we do not need

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventSummaryCreator.java (hunk 1 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventSummaryCreator.java:11]: it is ccd restriction it is in the incident.

**Review by apereverzin — APPROVED** [review]

**Pull request merged**